src/install/install.zig

```diff
@@ -3432,7 +3398,18 @@ pub const PackageManager = struct {
                 i,
                 &dependency,
                 resolution,
-            ) catch {};
+            ) catch |err| {
+                this.log.addZigErrorWithNote(
```

Comment:
This will cause failing optional dependencies to fail the build and it should not 

Instead, I suggest setting level to warn if the dependency.behavior is optional or peer dependency